static/js/syntaxModesRules-en.ts

```diff
@@ -0,0 +1,355 @@
+      syntaxRules.rule_ifElseOneLine(_IF,_ELSE,_IS,_IN),
+      syntaxRules.rule_expressions(_AT,_RANDOM),
+      syntaxRules.rule_repeat(_REPEAT,_TIMES),
+    ),
+  },
+  {
+    // Adds arithmetic
+    name: 'level6',
+    rules: syntaxRules.pipe(syntaxRules.baseRules(_AT,_RANDOM),
+      syntaxRules.rule_printSpace(_PRINT),
+      syntaxRules.rule_isAsk(_IS,_ASK),
+      syntaxRules.rule_is(_IS),
+      syntaxRules.rule_ifElseOneLine(_IF,_ELSE,_IS,_IN),
+      syntaxRules.rule_expressions(_AT,_RANDOM),
+      syntaxRules.rule_repeat(_REPEAT,_TIMES),
+      syntaxRules.rule_arithmetic(),
+    ),
+  },
+  {
+    // Adds indented blocks -- no changes to highlighter necessary
+    name: 'level7',
+    rules: syntaxRules.pipe(syntaxRules.baseRules(_AT,_RANDOM),
+      syntaxRules.rule_printSpace(_PRINT),
+      syntaxRules.rule_isAsk(_IS,_ASK),
+      syntaxRules.rule_is(_IS),
+      syntaxRules.rule_ifElse(_IF,_ELSE,_IS,_IN),
+      syntaxRules.rule_expressions(_AT,_RANDOM),
+      syntaxRules.rule_repeat(_REPEAT,_TIMES),
+      syntaxRules.rule_arithmetic(),
+    ),
+  },
+  {
+    // Replaces 'repeat' with 'for'
+    name: 'level8',
+    rules: syntaxRules.pipe(syntaxRules.baseRules(_AT,_RANDOM),
+      syntaxRules.rule_printSpace(_PRINT),
+      syntaxRules.rule_isAsk(_IS,_ASK),
+      syntaxRules.rule_is(_IS),
+      syntaxRules.rule_ifElse(_IF,_ELSE,_IS,_IN),
+      syntaxRules.rule_expressions(_AT,_RANDOM),
+      syntaxRules.rule_arithmetic(),
+      syntaxRules.rule_for(_FOR,_IN)
+    ),
+  },
+  {
+    // Replaces 'repeat' with 'for'
+    name: 'level9and10',
+    rules: syntaxRules.pipe(syntaxRules.baseRules(_AT,_RANDOM),
+    syntaxRules.rule_printSpace(_PRINT),
+    syntaxRules.rule_isAsk(_IS,_ASK),
+    syntaxRules.rule_is(_IS),
+    syntaxRules.rule_ifElse(_IF,_ELSE,_IS,_IN),
+    syntaxRules.rule_expressions(_AT,_RANDOM),
+    syntaxRules.rule_arithmetic(),
+    syntaxRules.rule_forRange(_FOR,_IN,_RANGE),
+    syntaxRules.rule_for(_FOR,_IN)
+    ),
+  },
+  {
+    // Nesting of 'for' loops (no changes necessary)
+    name: 'level11',
+    rules: syntaxRules.pipe(syntaxRules.baseRules(_AT,_RANDOM),
+      syntaxRules.rule_printSpace(_PRINT),
+      syntaxRules.rule_isAsk(_IS,_ASK),
+      syntaxRules.rule_is(_IS),
+      syntaxRules.rule_ifElse(_IF,_ELSE,_IS,_IN),
+      syntaxRules.rule_expressions(_AT,_RANDOM),
+      syntaxRules.rule_arithmetic(),
+      syntaxRules.rule_forRange(_FOR,_IN,_RANGE),
+    ),
+  },
+  {
+    // Adding fncall parens
+    name: 'level12',
+    rules: syntaxRules.pipe(syntaxRules.baseRules(_AT,_RANDOM),
+      syntaxRules.rule_printParen(_PRINT),
+      syntaxRules.rule_isInputParen(_IS,_INPUT),
+      syntaxRules.rule_is(_IS),
+      syntaxRules.rule_ifElse(_IF,_ELSE,_IS,_IN),
+      syntaxRules.rule_expressions(_AT,_RANDOM),
+      syntaxRules.rule_arithmetic(),
+      syntaxRules.rule_forRangeParen(_FOR, _IN,_RANGE),
+    ),
+  },
+// ----------------------------------------------------------------
```

Comment:
Are we ready to merge this? :) I guess the lang changes are finished further than 12?